packages/frontend/src/components/wallet-migration/metrics.js

```diff
@@ -1,63 +1,65 @@
-import Analytics from 'analytics-node';
+// import Analytics from 'analytics-node';
+import * as rudderanalytics from 'rudder-sdk-js';
 
 import { IS_MAINNET } from '../../config';
 import { KEY_ACTIVE_ACCOUNT_ID } from '../../utils/wallet';
 
-let segment = null;
+export let rudderAnalyticsReady = false;
+const DATA_PLANE_URL = 'https://nearpavelsqp.dataplane.rudderstack.com';
 
 export const initSegment = () => {
-    if (segment) {
-        return; // already initialized
-    }
-  
-    const segmentKey = IS_MAINNET
-        ? '3EUjxEBOReF6AWiwz1sAXhdyhn38x2a0'
-        : 'DSgndJjJhdaQ2GXTKf3VaZzXwDrJyj2d';
-
-    try {
-        segment = new Analytics(segmentKey);
-    } catch (e) {
-        console.error(e);
-    }
+    return new Promise((resolve) => {
+        if (rudderAnalyticsReady) {
+            return;
+        }
+      
+        const rudderAnalyticsKey = IS_MAINNET
+            ? '2RPBJoVn6SRht1E7FnPEZLcpNVh'
+            : '2RPBBhH1Dka2bYG6HKQmvkzNUdz';
+    
+        rudderanalytics.load(rudderAnalyticsKey, DATA_PLANE_URL);
+        rudderanalytics.ready(() => {
+            rudderAnalyticsReady = true;
+            resolve();
+        }); 
+    });
 };
 
 export const recordWalletMigrationEvent = (eventLabel, properties = {}) => {
-    if (!segment) {
+    if (!rudderAnalyticsReady) {
         return;
     }
+
     try {
         const accountId = localStorage.getItem(KEY_ACTIVE_ACCOUNT_ID);
-        segment.track({
-            userId: accountId,
-            event: eventLabel,
-            properties,
-        });
+        rudderanalytics.track(eventLabel, { ...properties, userId: accountId });
     } catch (e) {
         console.error(e);
     }
 };
 
 export const recordWalletMigrationState = (traits = {}, fallBackAccountId) => {
-    if (!segment) {
+    if (!rudderAnalyticsReady) {
         return;
     }
+
     try {
         const accountId = localStorage.getItem(KEY_ACTIVE_ACCOUNT_ID) || fallBackAccountId;
-        segment.identify({
-            userId: accountId,
-            traits: {
+        rudderanalytics.identify(
+            accountId,
+            {
                 ...traits,
                 userAgent: navigator?.userAgent || 'Unknown',
-            }
-        });
+            },
+        );
     } catch (e) {
         console.error(e);
     }
 };
 
 export const flushEvents = () => {
-    if (!segment) {
+    if (!rudderAnalyticsReady) {
         return;
     }
-    return segment.flush();
+    return rudderanalytics.reset();
```

Comment:
`rudder-stack-js` does not expose a publicly available method similar to the functionality of `segment.flush()` which is intended for forcing all queued events to be sent immediately to the server. `.reset()` does not do that here though.

 It simply resets the anonymousId as well as the userAccountId that we are tracking as per the implementation [here](https://github.com/rudderlabs/rudder-sdk-js/blob/66c998ba2785cd33445119f9468d0d3488b34f38/src/core/analytics.js#L1071).
 
I think we can rename this method to `resetUserState()` instead.